script/DeployL2OFTAdapter.s.sol

```diff
@@ -0,0 +1,163 @@
+            multiChainDeployer = ImmutableMultiChainDeployer(predictedAddress);
+        }
+
+        require(address(multiChainDeployer) == predictedAddress, "Prediction mismatch");
+
+        currentDeployment.multiChainDeployer = address(multiChainDeployer);
+
+        bytes32 proxySalt = createL2YnERC20UpgradeableProxySalt(msg.sender);
+        bytes32 implementationSalt = createL2YnERC20UpgradeableSalt(msg.sender);
+
+        address deployer = msg.sender;
+
+        address predictedERC20 = multiChainDeployer.getDeployed(proxySalt);
+        require(predictedERC20 == predictions.l2ERC20, "Prediction mismatch");
+
+        if (!isContract(currentDeployment.erc20Address)) {
+            vm.broadcast();
+            l2ERC20 = L2YnERC20Upgradeable(
+                multiChainDeployer.deployL2YnERC20(
+                    implementationSalt,
+                    proxySalt,
+                    baseInput.erc20Name,
+                    baseInput.erc20Symbol,
+                    deployer,
+                    deployer,
+                    type(L2YnERC20Upgradeable).creationCode
+                )
+            );
+
+            vm.broadcast();
+
+            address newOwner = getData(block.chainid).PROXY_ADMIN;
+            console.log("Changing owner for L2ERC20 to: %s", newOwner);
+            Ownable(getTransparentUpgradeableProxyAdminAddress(address(l2ERC20))).transferOwnership(newOwner);
+
+            console.log("Deployed L2ERC20 at: %s", address(l2ERC20));
+        } else {
+            l2ERC20 = L2YnERC20Upgradeable(currentDeployment.erc20Address);
+            console.log("Already deployed L2ERC20 at: %s", address(l2ERC20));
+        }
+
+        require(predictedERC20 == address(l2ERC20), "Prediction mismatch");
+
+        proxySalt = createL2YnOFTAdapterUpgradeableProxySalt(msg.sender);
+        implementationSalt = createL2YnOFTAdapterUpgradeableSalt(msg.sender);
+
+        address predictedOFTAdapter = multiChainDeployer.getDeployed(proxySalt);
+        require(predictedOFTAdapter == predictions.l2OFTAdapter, "Prediction mismatch");
+
+        if (!isContract(currentDeployment.oftAdapter)) {
+            vm.broadcast();
+            l2OFTAdapter = L2YnOFTAdapterUpgradeable(
+                multiChainDeployer.deployL2YnOFTAdapter(
+                    implementationSalt,
+                    proxySalt,
+                    address(l2ERC20),
+                    getData(block.chainid).LZ_ENDPOINT,
+                    deployer,
+                    deployer,
+                    type(L2YnOFTAdapterUpgradeable).creationCode
+                )
+            );
+
+            vm.broadcast();
+
+            address newOwner = getData(block.chainid).PROXY_ADMIN;
+            console.log("Changing owner for L2OFTAdapter to: %s", newOwner);
+            Ownable(getTransparentUpgradeableProxyAdminAddress(address(l2OFTAdapter))).transferOwnership(newOwner);
```

Comment:
Owned by a newly deployed timelock

The timelock should be fully owned and controlled by the security council what is now called PROXY_ADMIN in your script.